src/pages/index.js

```diff
@@ -6,57 +6,174 @@ import PopupWithImage from "../components/PopupWithImage";
+const editAvatarForm = editAvatarModalEl.querySelector(".modal__form");
 
 const editProfileModalEl = document.querySelector("#profile__edit-modal");
 const profileEditForm = editProfileModalEl.querySelector(".modal__form");
 
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "070eb85e-643e-4e8f-9d15-8b589c285fbd",
+    "Content-Type": "application/json",
+  },
+});
+
+let section;
+
+api.getInitialCards().then((cards) => {
+  console.log(cards);
+  section = new Section(
+    {
+      items: cards,
+      renderer: (data) => {
+        section.addItem(renderCard(data));
+      },
+    },
+    ".cards__list"
+    // cardsWrap.prepend('');
+  );
+  section.renderItems();
+});
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      userName: userData.name,
+      userDescription: userData.about,
+    });
+  })
+  .catch((e) => {
+    console.log(e);
+  });
+
 const userInfo = new UserInfo({
   userNameSelector: ".profile__title",
   userDescriptionSelector: ".profile__description",
 });
 
+const deleteConfirmation = new DeleteConfirmation({
+  popupSelector: "#delete-modal",
+  handleDeleteSubmit: (card, _id) => {
+    console.log("deleting this card =>", card);
+    // TODO: call the delete card API method
+
+    api.deleteCard(_id).then(() => {
+      card.remove();
+    });
+  },
+});
+deleteConfirmation.setEventListeners();
+
+const handleDeleteClick = (card, _id) => {
+  deleteConfirmation.open(card, _id);
+};
+
 console.log(userInfo.getUserInfo());
 
 const editProfileModal = new PopupWithForm({
   popupSelector: "#profile__edit-modal",
   //
   handleFormSubmit: (formValues) => {
     // use data from formValues instead of accessing .value directly
-    userInfo.setUserInfo({
-      userName: formValues.title,
-      userDescription: formValues.description,
-    });
+
+    // to update the avatar you could do something like this...
+    // .updateUserInfo({ avatar: "12345" })
+
+    // api.updateUserInfo({avatar: })
+    editProfileModal.updateButtonText("Saving...");
+    api
+      .updateUserInfo({ name: formValues.title, about: formValues.description })
+      .then((userData) => {
+        userInfo.setUserInfo({
+          userName: userData.name,
+          userDescription: userData.about,
+        });
+      })
+      .catch((e) => {
+        console.log(e);
+      })
+      .finally(() => {
+        editProfileModal.updateButtonText("Save");
+      });
+
     // move functionality for submit here
   },
 });
 editProfileModal.setEventListeners();
 
+const editAvatarModal = new PopupWithForm({
+  popupSelector: "#avatar__modal",
+  //
+  handleFormSubmit: (formValues) => {
+    editAvatarModal.updateButtonText("Saving...");
+    api
+      .editProfileInfo(formValues)
+      .then((userData) => {
+        userInfo.setUserInfo({
+          // finish setting the user info
+        });
```

Comment:
* I can't change the `avatar`. See the error https://snipboard.io/80wtZh.jpg  

Only methods of `UserInfo` should set/get the profile data (including the **avatar**)